CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/gui/JSingleChoiceComboBoxMenuNGTest.java

```diff
@@ -0,0 +1,252 @@
+     * Test the constructor works with a Text
+     */
+    @Test
+    public void testConstructor_WithText() {
+        final File iconFile = new File("non_existing_icon.png");
+        installedFileLocatorMockedStatic.when(() -> InstalledFileLocator.getDefault()
+                .locate("modules/ext/icons/drop_down_arrow.png", "au.gov.asd.tac.constellation.utilities", false))
+                .thenReturn(iconFile);
+
+        JSingleChoiceComboBoxMenu instance = new JSingleChoiceComboBoxMenu("Text", items);
+        String expResult = "Text";
+
+        String result = instance.getText();
+        assertEquals(result, expResult);
+    }
+
+    /**
+     * Test the constructor works with an Icon
+     */
+    @Test
+    public void testConstructor_WithIcon() {
+        final Icon icon = new ImageIcon(COOKIE_ICON_PATH);
+        JSingleChoiceComboBoxMenu instance = new JSingleChoiceComboBoxMenu(icon, items);
+        Icon result = instance.getIcon();
+        assertNotNull(result);
+    }
+
+    /**
+     * Test of getText method, of class JSingleChoiceComboBoxMenu.
+     */
+    @Test
+    public void testGetText() {
+        System.out.println("getText");
+        String text = "Test text";
+        JSingleChoiceComboBoxMenu instance = new JSingleChoiceComboBoxMenu(text, items);
+        String result = instance.getText();
+        assertEquals(result, text);
+    }
+
+    /**
+     * Test of setText method, of class JSingleChoiceComboBoxMenu.
+     */
+    @Test
+    public void testSetText() {
+        System.out.println("setText");
+        String text = "Test text";
+        JSingleChoiceComboBoxMenu instance = new JSingleChoiceComboBoxMenu("Text", items);
+        instance.setText(text);
+        String result = instance.getText();
+        assertEquals(result, text);
+    }
+
+    /**
+     * Test of getIcon method, of class JSingleChoiceComboBoxMenu.
+     */
+    @Test
+    public void testGetIcon() {
+        System.out.println("getIcon");
+        JSingleChoiceComboBoxMenu instance = new JSingleChoiceComboBoxMenu("Text", items);
+        Icon result = instance.getIcon();
+        assertNotNull(result);
+    }
+
+    /**
+     * Test of setIcon method, of class JSingleChoiceComboBoxMenu.
+     */
+    @Test
+    public void testSetIcon() {
+        System.out.println("setIcon");
+
+        JSingleChoiceComboBoxMenu instance = new JSingleChoiceComboBoxMenu("Text", items);
+
+        try {
+            final Icon newIcon1 = new ImageIcon(Utilities.toURI(iconFile).toURL());
+            final Icon newIcon2 = new ImageIcon(DROP_DOWN_ARROW_ICON_PATH);
+            instance.setIcon(newIcon1);
+            Icon result = instance.getIcon();
+            assertEquals(newIcon1, result);
+
+            instance.setIcon(newIcon2);
+            result = instance.getIcon();
+            assertEquals(newIcon2, result);
+        } catch (MalformedURLException ex) {
+            System.out.println("Could not create drop down arrow image");
+        }
```

Comment:
Throw the exception rather than catching it (probably want the unit test to fail in this case anyway)

Reply:
Removed the try catch block.